### PR TITLE
[wasm] Enable the full gc by default

### PR DESCRIPTION
**Pull request by lewing**

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

**Comment by lewing** [comment]
cc @pranavkm @SteveSandersonMS

**Comment by pranavkm** [comment]
Since I'm unfamiliar with this area, what does this change do?

**Comment by lewing** [comment]
It allows the wasm GC to collect regularly instead of only when control returns to JS.  The feature has been working for quite a while but had been off by default initially to avoid added risk.  For net5.0 we plan to enable it by default, it can be disabled by calling `mono_wasm_enable_on_demand_gc(0)`

**Review by marek-safar — APPROVED** [review]

File: eng/testing/WasmRunnerTemplate.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 dotnet xharness wasm test --engine=$JAVASCRIPT_ENGINE \
     --js-file=runtime.js -v \
     --output-directory=$XHARNESS_OUT \
-    -- --enable-gc --run WasmTestRunner.dll ${@:2}
```
Comment (akoeplinger) [inline eng/testing/WasmRunnerTemplate.sh:13]: this moved to tests.mobile.targets in master, you'll need to make the change there to resolve the merge conflict.

File: src/mono/wasm/runtime-test.js [hunk 2]
```diff
@@ -127,8 +127,8 @@ while (true) {
 		var arg = args [0].substring ("--runtime-arg=".length);
 		runtime_args.push (arg);
 		args = args.slice (1);
-	} else if (args [0] == "--enable-gc") {
-		enable_gc = true;
+	} else if (args [0] == "--disable-gc") {
```
Comment (akoeplinger) [inline src/mono/wasm/runtime-test.js:130]: I wonder if we should rename this to `--disable-ondemand-gc`, cause it doesn't really "disable" the GC completely like the name would suggest.

File: src/mono/wasm/runtime-test.js [hunk 3]
```diff
@@ -201,9 +201,9 @@ var Module = {
 			}
 		}
 
-		if (enable_gc) {
-			var f = Module.cwrap ('mono_wasm_enable_on_demand_gc', 'void', []);
-			f ();
+		if (!enable_gc) {
+			var f = Module.cwrap ('mono_wasm_enable_on_demand_gc', 'void', ['number']);
+			f (enable_gc);
```
Comment (akoeplinger) [inline src/mono/wasm/runtime-test.js:206]: this is a bit confusing, we already know that enable_gc can only be 0 in here. I'd prefer keeping the variable as a bool and passing 0 explicitly here.

**Pull request merged**